### PR TITLE
Improve DCPR request edit page

### LINKED ISSUE #182: Improve style of the DCPR request edit page
The DCPR request edit page needs some polish. Please improve the following:

- Placement of the `add another dataset` and `remove previous dataset` buttons
- The `Spatial extent` field should show the actual extent of the rectangle drawn on the map (it does show it when the user interacts with the map)
- There should be a validation of the `capture start date` and `capture end date` fields to ensure that the end is not defined to a date that is before the start

### PR DESCRIPTION
**Pull request by Fanevanjanahary**
fixes #182
- [x] Add validation error into the field capture end date
     - create page
image
     - edit page
     
image

- [x] Changed _**remove the previous DCPR request**_ button and added the remove button on each header of the DCPR request 
remove_dcpr

### CODE REVIEW

File: ckanext/dalrrd_emc_dcpr/logic/action/dcpr/create.py
Comment (ricardogsilva) [inline ckanext/dalrrd_emc_dcpr/logic/action/dcpr/create.py:86]: Please improve the wording here, as the proposed change is not very clear

File: ckanext/dalrrd_emc_dcpr/logic/action/dcpr/update.py
Comment (ricardogsilva) [inline ckanext/dalrrd_emc_dcpr/logic/action/dcpr/update.py:31]: As per the above comment, please improve the wording of the error

File: ckanext/dalrrd_emc_dcpr/assets/js/removeDcprRequestDatasets.js
Comment (ricardogsilva) [inline ckanext/dalrrd_emc_dcpr/assets/js/removeDcprRequestDatasets.js:3]: please fix the typo on the name of the module

```suggestion
ckan.module('removeDcprRequestDatasets', function(jQuery, _){
```

File: ckanext/dalrrd_emc_dcpr/templates/dcpr/snippets/request_form.html
Comment (ricardogsilva) [inline ckanext/dalrrd_emc_dcpr/templates/dcpr/snippets/request_form.html:118]: remember to update the `data-module` value after you fix the typo in the name of the module (as mentioned in a comment above)

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
Nice work!

However, there are some things which should be changed, as mentioned in the inline comments.

In addition to those changes:

The inclusion of individual remove buttons for each dataset is a nice addition. However please ensure that there is always at least one dataset for a DCPR request. This means that the user must not be able to delete all datasets.

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
you took care of the inline comments, which is good :smile:

But you still need to modify your js code in order to ensure it is not possible for the user to remove a dataset when it is the only one present in the request. 

It is currently possible to save a DCPR request that doesn't contain any datasets. We don't want to allow that. The previous implementation did the correct thing, so this is a regression. Please fix it before we merge this.

File: ckanext/dalrrd_emc_dcpr/templates/dcpr/snippets/request_form.html
Comment (Fanevanjanahary) [inline ckanext/dalrrd_emc_dcpr/templates/dcpr/snippets/request_form.html:121]: Hi, @ricardogsilva I think I added here the check the number of the dataset.  The button is disabled it is the only dataset

Comment (ricardogsilva) [reply ckanext/dalrrd_emc_dcpr/templates/dcpr/snippets/request_form.html:121]: It doesn't seem to work 100%, at least not on my local testing. I am able to go and edit an existing DCPR request, then delete all of its current datasets, then save and it works without any error.

Please double check on your side that it works OK when creating or updating DCPR requests and let me know. I will re-check after that

Comment (Fanevanjanahary) [reply ckanext/dalrrd_emc_dcpr/templates/dcpr/snippets/request_form.html]: Hi @ricardogsilva updated, thank you for the review

**Comment by ricardogsilva** [comment]
@Fanevanjanahary , when you eventually get back to this, please request a review from @Mohab25 rather than myself. Thanks!

**Pull request merged**
